build_membership.py

```diff
@@ -0,0 +1,225 @@
+
+    # These modules do not exist pre-Qt 5,
+    # so do not bother testing for them.
+    __all__.remove("QtSql")
+    __all__.remove("QtSvg")
+
+    from PySide2 import *
+
+    # Serialise members
+    members = {}
+    for name, module in locals().copy().items():
+        if name.startswith("_"):
+            continue
+
+        if name in ("json", "members"):
+            continue
+
+        members[name] = dir(module)
+
+    # Write to disk
+    with open("reference_members.json", "w") as f:
+        json.dump(members, f, indent=4)
+
+
+def build_tests():
+    """Build membership tests
+
+    Members only available in Qt 5 are excluded, along with member
+    exclusive to a paricular binding.
+
+    """
+
+    header = """\
+import os
+import sys
+import json
+
+with open("reference_members.json") as f:
+    reference_members = json.load(f)
+
+excluded = {excluded}
+
+""".format(excluded=json.dumps(excluded, indent=4))
+
+    test = """\
+def test_{binding}_members():
+    os.environ["QT_PREFERRED_BINDING"] = "{Binding}"
+
+    if "PyQt" in "{Binding}":
+        # PyQt4 and 5 performs some magic here
+        # that must take place before attempting
+        # to import with wildcard.
+        from Qt import Qt as _
+
+    if "PySide2" == "{Binding}":
+        # PySide2, as of this writing, doesn't include
+        # these modules in it's __all__ list; leaving
+        # the wildcard import below untrue.
+        from Qt import __all__
+        for missing in ("QtWidgets",
+                        "QtXml",
+                        "QtHelp",
+                        "QtPrintSupport"):
+            __all__.append(missing)
+
+    from Qt import *
+
+    target_members = dict()
+    for name, module in locals().copy().items():
+        if name.startswith("_"):
+            continue
+
+        target_members[name] = dir(module)
+
+    missing = dict()
+    for module, members in reference_members.items():
+        for member in members:
+
+            # Ignore those that have no Qt 4-equivalent.
+            if member in excluded.get(module, []):
+                continue
+
+            if member not in target_members.get(module, []):
+                if module not in missing:
+                    missing[module] = []
+                missing[module].append(member)
+
+    message = ""
+    for module, members in missing.items():
+        message += "\\n%s: \\n - %s" % (module, "\\n - ".join(members))
+
+    assert not missing, "{Binding} is missing members: %s" % message
+
+"""
+
+    tests = list(test.format(Binding=binding,
+                             binding=binding.lower())
+                 for binding in ["PyQt5",
+                                 "PyQt4",
+                                 "PySide"])
+
+    with open("test_membership.py", "w") as f:
+        contents = header + "\n".join(tests)
+        print(contents)  # Preview content during tests
+        f.write(contents)
+
+
+# Don't consider these members
```

Comment:
Would you mind expanding this comment on why these are not considered members?

Is it because they are either internals (__something__) or binding-specific?